eng/common/TestResources/New-TestResources.ps1

```diff
@@ -313,25 +311,53 @@ foreach ($templateFile in $templateFiles) {
         }
     }
 
-    foreach ($key in $deploymentOutputs.Keys)
+    if ($File)
     {
-        $value = $deploymentOutputs[$key]
-        
-        if ($CI) {
-            # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
-            # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
-            Write-Host "Setting variable '$key': ***"
-            Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($value)"
-            Write-Host "##vso[task.setvariable variable=$key;]$($value)"
-        } else {
-            Write-Host ($shellExportFormat -f $key, $value)
+        if (!($IsWindows))
+        {
+            Write-Host "File option is supported only on Windows"
         }
+
+        $outputPath = Join-Path $env:USERPROFILE ".Azure" "TestEnvironments"
+        $outputFile = Join-Path $outputPath $serviceName
+
+        $environmentText = $deploymentOutputs | ConvertTo-Json;
+        $bytes = ([System.Text.Encoding]::UTF8).GetBytes($environmentText)
+        $protectedBytes = [Security.Cryptography.ProtectedData]::Protect($bytes, $null, [Security.Cryptography.DataProtectionScope]::LocalMachine)
+
+        New-Item -Type Directory $outputPath -Force | Out-Null
```

Comment:
Unnecessary. `Set-Content` will create the file. Piping to `Out-Null` is also slower (tip). Just assign to `$null`:
```powershell
$null = do-whatever
```

Reply:
> Set-Content will create the file

But it won't create a directory.

Reply:
Yeah, I forgot that. Ran into it last night with a script I was writing. Used to xcopy/robocopy just doing the right thing. :)